applications/app/controllers/OptInController.scala

```diff
@@ -6,16 +6,28 @@ import play.api.mvc.{Action, Controller, Cookie, DiscardingCookie}
 
 import scala.concurrent.duration._
 
-object OptInController extends Controller {
-  case class OptInFeature(cookieName: String, lifetime: Int = 90.days.toSeconds.toInt) {
-    def opt(choice: String) = choice match {
-      case "in" => optIn()
-      case _ => optOut()
-    }
-
-    def optIn() = SeeOther("/").withCookies(Cookie(cookieName, "true", maxAge = Some(lifetime)))
-    def optOut() = SeeOther("/").discardingCookies(DiscardingCookie(cookieName))
+trait OptInOutFeature extends Controller {
```

Comment:
depends how you look at it! after all https is default now, so maybe you're opting into insecure content?  either way I feel it's more a naming/terminology problem than a problem needing more complexity in the code, but I appreciate there's urls to consider too.